sklearn/naive_bayes.py

```diff
@@ -949,3 +988,258 @@ def _joint_log_likelihood(self, X):
         jll += self.class_log_prior_ + neg_prob.sum(axis=1)
 
         return jll
+
+
+class CategoricalNB(BaseDiscreteNB):
+    """Naive Bayes classifier for categorical features
+
+    The categorical Naive Bayes classifier is suitable for classification with
+    discrete features, that are categorically distributed. The categories of
+    each feature are drawn from a categorical distribution.
+
+    Read more in the :ref:`User Guide <categorical_naive_bayes>`.
+
+    Parameters
+    ----------
+    alpha : float, optional (default=1.0)
+        Additive (Laplace/Lidstone) smoothing parameter
+        (0 for no smoothing).
+
+    fit_prior : boolean, optional (default=True)
+        Whether to learn class prior probabilities or not.
+        If false, a uniform prior will be used.
+
+    class_prior : array-like, size (n_classes,), optional (default=None)
+        Prior probabilities of the classes. If specified the priors are not
+        adjusted according to the data.
+
+    handle_unknown : String, (default='warn')
+        Can be 'ignore', 'warn' or 'error'. Determines the behaviour of the
+        classifier, if it encounters unseen categories in the prediction step.
+
+    Attributes
+    ----------
+    class_log_prior_ : array, shape (n_classes, )
+        Smoothed empirical log probability for each class.
+
+    feature_log_prob_ : array, shape (n_classes, n_features)
+        Empirical log probability of features
+        given a class, ``P(x_i|y)``.
+
+    class_count_ : array, shape (n_classes,)
+        Number of samples encountered for each class during fitting. This
+        value is weighted by the sample weight when provided.
+
+    category_count_ : dict of arrays, len n_features
+        Holds arrays of shape (n_classes, n_categories of respective feature)
+        for each feature. Each array provides the number of samples
+        encountered for each class and category of the specific feature.
+
+    n_features_ : int
+        Number of features of each sample.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> X = np.random.randint(5, size=(6, 100))
+    >>> y = np.array([1, 2, 3, 4, 5, 6])
+    >>> from sklearn.naive_bayes import CategoricalNB
+    >>> clf = CategoricalNB()
+    >>> clf.fit(X, y)
+    CategoricalNB(alpha=1.0, class_prior=None, fit_prior=True,
+                  handle_unknown='warn')
+    >>> print(clf.predict(X[2:3]))
+    [3]
+    """
+
+    old_numpy_ = parse_version(np.__version__) < parse_version('1.9.0')
+
+    def __init__(self, alpha=1.0, fit_prior=True, class_prior=None,
+                 handle_unknown='warn'):
+        if CategoricalNB.old_numpy_:
```

Comment:
I don't think this should be a public attribute. Make it `_old_numpy`

Reply:
I also don't think it needs to be an attribute of CategoricalNB. Just make it a module-level global.